### ISSUE TITLE
Add new event type for popup `hover-and-click`

### ISSUE THREAD

**Issue by DevOfManyThings** [skeletonlabs/skeleton#1327]
### Describe what feature you'd like. Pseudo-code, mockups, or screenshots of similar solutions are encouraged!

The new event type `hover-and-click` to [popup](https://github.com/skeletonlabs/skeleton/blob/master/src/lib/utilities/Popup/popup.ts#L234)

When a user hovers over the target element, it should appear, and if they hover away it should disappear.
If a user clicks on the target element, it should appear, if they click again after it's visible it should disappear (this is to support mobile users)

Would need to test it, but I _think_ it should be roughly adding this:
`if (args.event === 'hover-and-click') {
		node.addEventListener('mouseover', show, true);
		node.addEventListener('mouseout', close, true);
		window.addEventListener('click', onWindowClick, true);
	}`

This is mostly relevant to tooltips, as interactive modals won't work because of the `mouseout` event.

Along with the enhancement, it is likely also worth updating the documentation to include an example for this (as this would be the best way to support tooltips generically), and minor rephrasing of the documentation as discussed in the original issue linked below.

### What type of pull request would this be?

Enhancement

### Any links to similar examples or other references we should review?

Earlier discussion with Chris here: https://github.com/skeletonlabs/skeleton/issues/1316

**Comment by endigo9740** [comment]
@DevOfManyThings FYI I've just completed my refactor of the popup feature per:
- https://github.com/skeletonlabs/skeleton/issues/1350

Review my comment on the changes here:
- https://github.com/skeletonlabs/skeleton/issues/1350#issuecomment-1533784403

I've given your above request consideration while implemented these changes, and opted to forego the dedicated event for this. Instead what I've done is simplify the events we do support, which include:

1. Click - click to open, click off to close
2. Hover - hover over to open, hover off to close
3. Focus-Blur - open on focus, close on blur
4. Focus-Click - open on focus, click off to close

You specified your goal above was to make it easier to use hover-able items on mobile. However, I don't believe this is a UX pattern we want to encourage. I've provided a slightly updated recommendation for this on the updated documentation:

![Screenshot 2023-05-03 at 5 43 55 PM](https://user-images.githubusercontent.com/1509726/236066802-8c8ddc20-c7b2-484b-8c0d-3b8c706d5040.png)

In short, I think hover of any kind is bad for mobile users. That you should avoid putting contents in hover-able popups that must be visible on mobile too. There's a great precedence for this baked into browsers - which is the `title` attribute. By default this works on mobile when you hover and element. However on mobile, this is just disregarded. Same idea here.

In short, we're not going to be moving forward with this. But I appreciate the suggestion!

**Comment by DevOfManyThings** [comment]
@endigo9740 Hey Chris, sorry I didn't get a chance to implement this but with the refactor that might have been for the best!

To clarify my focus was on being able to have a popup that shows on hover for desktop users, but can be clicked/tapped on by mobile users and still show.
Perhaps a better name for that would have been `hover-or-click`, as it would show whether the user clicks or hovers, and also disappear when the user clicks or hovers away.

For desktop users it's basically just the hover events that will fire (can't click off of it as you need to hover away first), and for mobile users it's basically just the click events (as hover isn't well supported), but from a developer perspective it gets you hover support on desktop without losing mobile support.

The primary usecase for this would be tooltips that the user won't be interacting with, as it's a smoother user experience for tooltips to show on hover for desktop users, but mobile users clicking is required.

Basically trying to do the equivalent of:
```
<button
	class="btn variant-filled-primary"
	use:popup={{ event: 'click', target: "focus", placement: 'top' }}
	use:popup={{ event: 'hover', target: "focus", placement: 'top' }}
>
	Sign in
</button>
```

I'll actually have time to make this tomorrow if you'd be interested in me making a PR towards your new branch? Or are you just wanting to keep it simple for now?

**Comment by endigo9740** [comment]
Sounds great in theory @DevOfManyThings but I don't really know of a clean way to tell the Action when to switch - what's mobile and what's desktop. That line is very much blurred these days. You have low resolution computers and high resolution mobile devices. You could try to determine if touch is available (maybe? I'm not aware of a great way to do this), but then you might end up with folks with touch screen desktop monitors not seeing hover tooltips. It's a bit of a rabbit hole.

I think at this point we're better to let end users decide this for themselves and modify the popupSetttings `event` type at their own discretion. They'll know their audience better than we will.

**Comment by DevOfManyThings** [comment]
@endigo9740 I mean honestly what I was thinking of just literally adding both the event listeners for both click and hover, which should _just work_? No need to determine whether the current user is desktop/mobile or supports anything.

```
case 'hover-or-click':
	triggerNode.addEventListener('click', toggle, true);
	window.addEventListener('click', onWindowClick, true);
	triggerNode.addEventListener('mouseover', open, true);
	triggerNode.addEventListener('mouseleave', close, true);
	break;
```

**Scenario 1: Desktop**
User hovers over element, `mouseover` fires so popup is displayed. If they click on it, `click` is triggered, but it just remains on screen so no impact.

When they hover away, `mouseleave` fires, which fires the below `destroy()` function, so `onWindowClick` can never be fired (if they do somehow trigger it the popup just closes anyway so desired behaviour still happens).

**Scenario 2: Mobile**
Mobile user can't hover over elements (if they can, they'll function the same as scenario 1).
They click/tap on the element which fires the `click` event, causing popup to render. And they just need to tap off the element causing the `onWindowClick` event to fire which closes the popup.
The `hover` event listeners are wasted in this scenario, but I don't think that will have any real world performance impact.

This re-uses existing functions/methods, just adding in an extra case (and documentation), so as such is already cleaned up in the provided `destroy()` for this component.

I'll raise the PR after I test it tomorrow if the above seems generally fine :)

**Comment by endigo9740** [comment]
@DevOfManyThings hmm, not sure I particularly like that to be honest. Event listeners have a cost, and this is essentially doubling that of the other events. I'm worried this is still missing the forest for the trees though. Hover elements have their place, but I think the better practice is to use them sparingly and with the understanding they won't translate to mobile. In other words, functional parts of your UI should be be placed inside a hovered popup. I always think of them like styled replacements for the title attribute.

Per today's updates users can actually do what you're saying by doubling the `use:action` on the trigger with different settings. I feel very redundant, but it will be technically possible now. Perhaps we just give that a go for now and listen to feedback. If a lot of folks are clamoring for fallback support for hover states when we review other ideas?
